mlabs/src/Mlabs/NFT/Governance/Validation.hs

```diff
@@ -36,19 +77,282 @@ instance ValidatorTypes GovManage where
 
 -- | Minting policy for GOV and xGOV tokens.
 mkGovMintPolicy :: NftAppInstance -> GovAct -> ScriptContext -> Bool
-mkGovMintPolicy _ act _ =
+mkGovMintPolicy appInstance act ctx =
   case act of
     InitialiseGov ->
-      True
+      traceIfFalse "NFT-Gov-Mint: Only allowed to mint a proof token" checkOnlyProofMinted
+        && traceIfFalse "NFT-Gov-Mint: uniquetoken and proofToken are not sent to the right address" checkProofTokenToRightAddress
+        && traceIfFalse "NFT-Gov-Mint: The unique token is not present" checkUniqueTokenIsPresent
     MintGov ->
```

Comment:
`Head` must be present in this transaction, see previous comment about determining the correct `CurrencySymbol`.